File: app/src/main/java/io/memetic/alcohdex/core/BaseViewModel.java
package io.memetic.alcohdex.core;

import android.content.Context;
import android.content.res.Resources;
import android.databinding.BaseObservable;

import javax.inject.Inject;

import io.memetic.alcohdex.ComponentRegistry;
import io.memetic.alcohdex.core.interfaces.ViewModel;

/**
 * TODO class description
 *
 * @author manuel.silva@mobileforming.com
 *         4/7/17
 */
public abstract class BaseViewModel extends BaseObservable implements ViewModel {
    @Inject
    Context mContext;
    @Inject
    Resources mResources;

    {
        ComponentRegistry.getInstance().getAppComponent().inject(this);
    }

    public Context getContext() {
        return mContext;
    }

    public Resources getResources() {
        return mResources;
    }
}

File: app/src/main/java/io/memetic/alcohdex/core/BaseViewModel.kt
package io.memetic.alcohdex.core

import android.content.Context
import android.content.res.Resources
import android.databinding.BaseObservable
import io.memetic.alcohdex.ComponentRegistry
import io.memetic.alcohdex.core.interfaces.ViewModel
import javax.inject.Inject

abstract class BaseViewModel : BaseObservable(), ViewModel {
    lateinit var context: Context
        @Inject internal set

    lateinit var resources: Resources
        @Inject internal set

    init {
        ComponentRegistry.getInstance().appComponent.inject(this)
    }
}
